VRDR.Messaging/ParameterUtils.cs

```diff
@@ -0,0 +1,358 @@
+                foreach (Parameters.ParameterComponent item in g.Part)
+                {
+                    var value = item.Value;
+                    if (item.Name == field)
+                    {
+                        if(s != null && s.Length > 0){ // null values or empty strings should not be added
+                            list.Add(Tuple.Create(item.Name,(Base) new FhirString(s)));
+                        }
+                    }else{
+                            list.Add(Tuple.Create(item.Name,(Base) value));
+                    }
+                }
+            }
+            // If list doesn't include field, add it with the value
+            var element = list.Find(elem => elem.Item1 == field);
+            if (element == null)
+            {
+                var part = Tuple.Create(field, (Base)new FhirString(s));
+                list.Add(part);
+            }
+            // If the group exists, remove it.
+            if (g != null)
+            {
+                Record.Remove(group);
+            }
+            // Insert the list as the new group
+            if(list.Count > 0){
+                Record.Add(group, list);
+            }
+        }
+        /// <summary>SetParameterString:   Get a String parameter associated with a top level parameter </summary>
+                private void SetParameterString(string s, string field)
+        {
+            SetParameterString(s, null, field);
+        }
+        /// <summary>GetParameterString:   Get a String parameter associated with a group (top level) and field (part) parameter </summary>
+        private string GetParameterString(string group, string field)
+        {
+            if (group != null)
+            {
+                var g = Record.GetSingle(group);
+                if (g == null){ // group doesn't exist
+                    return null;
+                }
+                var element = g.Part.Find(elem => elem.Name == field);
+                if ((element != null) && (element.Value != null))
+                {
+                    return ((FhirString)element.Value).Value;
+                }
+                else
+                {
+                    return null; // field in group doesn't exist
+                }
+            }
+            else
+            {  // group doesn't exist
+                return null;
+            }
+        }
+        /// <summary>GetParameterString:   Get a String parameter associated with a top level parameter </summary>
+        private string GetParameterString(string field)
+        {
+            Parameters.ParameterComponent f = Record.GetSingle(field);
+            if (f == null){
+                return null;
+            } else {
+                FhirString fs = (FhirString)f.Value;
+                string s = (string)fs.Value;
+                return s;
+            }
+        }
+        // These should be simply referenced from VRDR/Program.cs....Not sure why they aren't visible from there.
```

Comment:
The DictToCoding function in DeathRecord.cs is private, try making it public to use it here. It seems like more of a util function so I think its appropriate to make it public